starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -405,19 +418,25 @@ def to_python(self, value_types: List[dict], values: CairoData) -> NamedTuple:
         :param value_types: Types of values to be serialized
         :param values: Values to be serialized
         :return: tuple (full calldata, dict with all arguments with their Cairo representation)
+        :raises CairoSerializerException: when general error occurred
+        :raises InvalidDataException: when an error occurred while transforming a value
         """
         type_by_name = self._abi_to_types(value_types)
         initial_len = len(values)
 
         result = {}
         for name, cairo_type in type_by_name.items():
-            transformed, values = self.resolve_type(cairo_type).to_python(
-                cairo_type, name, values
-            )
+            try:
+                transformed, values = self.resolve_type(cairo_type).to_python(
+                    cairo_type, name, values
+                )
+            except ValueError as err:
+                raise InvalidDataException(str(err)) from err
```

Comment:
Why was this added? If all errors were replaced by custom ones, are these lines necessary?

Reply:
Yes, because not all of them were replaced. Felt still returns ValueError, because I didn't want to change any errors outside the module

Reply:
Are we sure only "default" exception that can be thrown in this module is `ValueError`? Perhaps this should be general `except Exception as err`?

Reply:
Yes, it is only the ValueError. Other errors are general python errors and we shouldn't catch them